pkg/network/ebpf/c/protocols/grpc/helpers.h

```diff
@@ -178,15 +177,20 @@ static __always_inline grpc_status_t is_grpc(const struct __sk_buff *skb, const
         info.data_off += current_frame.length;
     }
 
+    char content_type_buf[GRPC_CONTENT_TYPE_LEN];
+
 #pragma unroll(GRPC_MAX_FRAMES_TO_PROCESS)
-    for (__u8 i = 0; i < GRPC_MAX_FRAMES_TO_PROCESS && status == PAYLOAD_UNDETERMINED; ++i) {
+    for (__u8 i = 0; i < GRPC_MAX_FRAMES_TO_PROCESS; ++i) {
```

Comment:
Does the condition of status == PAYLOAD_UNDETERMINED is not relevant anymore for the iteration pf the frames to process? I mean is it possible for us to be here in a state which is not PAYLOAD_UNDETERMINED, such as PAYLOAD_NOT_GRPC?

Reply:
no, I just moved the comparison to be later in the process. As you can see, the only place that changes the value of `status` is inside this loop